idaes/models/unit_models/packed_bed_reactor.py

```diff
@@ -0,0 +1,337 @@
+            balance_type=self.config.energy_balance_type,
+            has_heat_of_reaction=self.config.has_heat_of_reaction,
+            has_heat_transfer=self.config.has_heat_transfer)
+
+        self.control_volume.add_momentum_balances(
+            balance_type=self.config.momentum_balance_type,
+            has_pressure_change=self.config.has_pressure_change)
+
+        self.control_volume.apply_transformation()
+
+        # Add Ports
+        self.add_inlet_port()
+        self.add_outlet_port()
+
+        # Set references to balance terms at unit level
+        add_object_reference(self, "length", self.control_volume.length)
+
+        # Add volume variable for full reactor
+        units = self.config.property_package.get_metadata()
+        self.volume = Var(initialize=1,
+                          doc="Reactor Volume",
+                          units=units.get_derived_units("volume"))
+
+        self.area = Var(initialize=1,
+                        doc="area",
+                        units=units.get_derived_units("area"))
+
+        self.geometry = Constraint(expr=self.volume == self.config.mass_catalyst/self.config.bulk_density)
+
+        self.area_from_volume = Constraint(expr=self.area == self.volume/self.length)
+
+        def calculate_void_fraction(self):
+            return 1 - self.config.bulk_density/self.config.catalyst_density
+        self.epsilon = Expression(rule=calculate_void_fraction)
+
+        self.real_area = Constraint(expr=self.control_volume.area == self.area*
+                                        self.epsilon)
+
+        # Add PBR performance equation
+        @self.Constraint(self.flowsheet().config.time,
+                         self.control_volume.length_domain,
+                         self.config.reaction_package.rate_reaction_idx,
+                         doc="PBR performance equation")
+        def performance_eqn(b, t, x, r):
+            if x == 0.0:
+               return Constraint.Skip
+            else:
+                return b.control_volume.rate_reaction_extent[t, x, r] == (
+                        b.control_volume.reactions[t, x].reaction_rate[r] *
+                        b.control_volume.area *(
+                        b.config.bulk_density /
+                        b.epsilon))
+
+        if (self.config.has_heat_transfer is True and
+                self.config.energy_balance_type != EnergyBalanceType.none):
+            self.heat_duty = Reference(self.control_volume.heat[...])
+        if (self.config.has_pressure_change is True and
+                self.config.momentum_balance_type != MomentumBalanceType.none):
+            self.deltaP = Reference(self.control_volume.deltaP[...])
+
+    def _get_performance_contents(self, time_point=0):
+        var_dict = {"Volume": self.volume}
+        var_dict = {"Length": self.length}
+        var_dict = {"Area": self.area}
+
+        return {"vars": var_dict}
+
+    def get_costing(self, year=None, module=costing, **kwargs):
```

Comment:
You can remove this `get_costing` stuff - it is in the process of being deprecated and will no longer work after November.